.github/workflows/upload-win-installer.yml

```diff
@@ -0,0 +1,96 @@
+name: Upload Windows Installer
+
+on:
+  release:
+    types: [created, published, edited]
```

Comment:
Ref: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

So my understanding is, the "active" workflow instructions will always be the ones from from `main`.   Over time, that may not necessarily match the requirements/expectations/behavior of the `contrib/win-installer` scripts - which will come from the release-branch.

One idea (would also partially address the hard-coded golang version issue I mentioned) could be to call a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) from the triggering branch.

Reply:
Thats right. I almost did it this way since its typically better to prefer the latest installer script (and it would match the default of the workflow_dispatch) The only reason I didn't was it would mean two job executions and some extra noise, and the other direct workflow mechanism is usable if we needed a newer installer.

That said I was very on the fence on this so happy to move it that way if there is a strong preference.

Reply:
I'm not 100% comfortable asking to make an already complex thing, MORE complex.  There is a kind of cheat-mode option here:  Document (somewhere) that only the latest installer from `main` is supported.

That leaves folks depending on older releases in a lurch, but (at least for now) keeps the automation simple.  Not sure how others feel about that idea.